aws/ec2metadata/token_provider.go

```diff
@@ -62,7 +62,7 @@ func (t *tokenProvider) fetchTokenHandler(r *request.Request) {
 
 			// Check if request timed out while waiting for response
 			if e, ok := requestFailureError.OrigErr().(awserr.Error); ok {
-				if e.Code() == "RequestError" {
+				if e.Code() == request.ErrCodeRequestError {
```

Comment:
This would be a breaking change for V1 SDK. But for V2 SDK, I think timeouts should be wrapped with "RequestTimeout" error instead of Request Error.

Reply:
Makes sense. A good timeout sorry would be helpful.